grammar.js

```diff
@@ -805,19 +805,25 @@ module.exports = grammar({
     // Primitives
     //
 
+    // Here we tolerate unescaped newlines in double-quoted and
+    // single-quoted string literals.
+    // This is legal in typescript as jsx/tsx attribute values (as of
+    // 2020), and perhaps will be valid in javascript as well in the
+    // future.
+    //
     string: $ => choice(
       seq(
         '"',
         repeat(choice(
-          token.immediate(prec(PREC.STRING, /[^"\\\n]+|\\\r?\n/)),
+          token.immediate(prec(PREC.STRING, /[^"\\\n]+|\\?\r?\n/)),
```

Comment:
I think that if we're allowing newlines, you could simplify the regex so that newlines are not explicitly excluded on the left-hand-side of the OR operator. Then, the right-hand-side would only be for handling escaped newlines.

Something like `[^"\\]|\\\r?\n`

Reply:
Good point. Also, I'm now noticing that the regexp doesn't accept escaped quotes as in `"\""` or `'\''`. Looking into this.

Reply:
Nevermind, `$.escape_sequence` takes care of escaped quotes.

Reply:
I simplified the regexp and added tests to cover my expectations. I also moved my original test for jsx attributes to `literals.txt` which seems more appropriate. `expressions.txt` still has some basic tests for strings and numbers which are probably fine to stay there.